framework/src/utils/RandomData.C

```diff
@@ -96,32 +96,29 @@ RandomData::updateGenerators()
   }
 
   if (_is_nodal)
-  {
-    MeshBase::const_node_iterator it = _rd_mesh.getMesh().active_nodes_begin();
-    MeshBase::const_node_iterator end_it = _rd_mesh.getMesh().active_nodes_end();
+    updateGeneratorHelper(_rd_mesh.getMesh().active_nodes_begin(), _rd_mesh.getMesh().active_nodes_end());
```

Comment:
Wow - I just really looked at the code for the first time... the current implementation is just not right.  Relying on iterating over the whole mesh in the same way on every processor just isn't right.

It shouldn't hold up this PR (this is definitely an improvement).  But this needs to get redesigned.